client/src/user-context.tsx

```diff
@@ -53,9 +85,23 @@ export function UserDataProvider(props: { children: React.ReactNode }) {
         // have to create a separate context for it.
         waffle: data.waffle,
       };
+    },
+    {
+      initialData: sessionStorageData() || undefined,
     }
   );
 
+  React.useEffect(() => {
+    if (data !== undefined) {
+      // But do you have it in sessionStorage?!
```

Comment:
Nit: We are not really testing whether it exists in `sessionStorage` and only adding if it does not. We are just adding it irrespective. I think that is fine, just saying the comment suggests we are doing something like `if (!sessionStorage.getItem(...) then sessionStorage.setItem`

Reply:
something like

```suggestion
      // ensure user data is also stored/updated in `sessionStorage`
```